api/cryptocom/requests.go

```diff
@@ -0,0 +1,178 @@
+package cryptocom
+
+import (
+	"database/sql"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/shopspring/decimal"
+)
+
+func (c *Client) AuthRequest() *Request {
+	r := &Request{
+		Id:     1,
+		Type:   AuthRequest,
+		Method: "public/auth",
+		ApiKey: c.key,
+		Nonce:  generateNonce(),
+	}
+
+	c.generateSignature(r)
+	return r
+}
+
+func (c *Client) subscribeRequest(channels []string) *Request {
+	return &Request{
+		Id:     1,
+		Type:   SubscribeRequest,
+		Method: "subscribe",
+		Params: map[string]interface{}{"channels": channels},
+		Nonce:  generateNonce(),
+	}
+}
+
+func (c *Client) hearBeatRequest(reqId int) *Request {
+	return &Request{
+		Id:     reqId,
+		Type:   HeartBeat,
+		Method: "public/respond-heartbeat",
+	}
+}
+
+func (c *Client) createOrderRequest(
+	reqID int,
+	ask string,
+	bid string,
+	orderSide string,
+	orderType string,
+	price decimal.Decimal,
+	volume decimal.Decimal,
+	uuid uuid.UUID) *Request {
+	var volumeKey string
+	var volumeValue decimal.Decimal
+	if strings.ToUpper(orderType) == "MARKET" && strings.ToUpper(orderSide) == "BUY" {
+		volumeKey = "notional"
+		volumeValue = volume.Mul(price)
```

Comment:
Let's just put here amount and make a comment, that for Market Buy order, Volume is notional